QueueImplementation/VM/CustomCollectionViewModel.cs

```diff
@@ -0,0 +1,363 @@
+		public int? CurrentElement
+		{
+			get => this.elementToProcess;
+			set
+			{
+				if (this.elementToProcess == value)
+					return;
+
+				this.elementToProcess = value;
+				this.RaisePropertyChanged(() => this.CurrentElement);
+			}
+		}
+
+		/// <summary>
+		/// Gets or sets the value indicating whether choice is still available.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <value><c>true</c> if the choice is still available; otherwise, <c>false</c>.</value>
+		public bool IsChoiseAvailable 
+		{
+			get
+			{
+				return this.isChoisePossible;
+			}
+			set
+			{
+				if (this.isChoisePossible == value)
+					return;
+
+				this.isChoisePossible = value;
+				this.RaisePropertyChanged(() => this.IsChoiseAvailable);
+			}
+		}
+
+		/// <summary>
+		/// Gets or sets the message.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <value>The message.</value>
+		public string Message
+		{
+			get
+			{
+				return this.message;
+			}
+			set
+			{
+				this.message = value;
+				this.RaisePropertyChanged(() => this.Message);
+			}
+		}
+
+		/// <summary>
+		/// Gets the remove command.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <value>The remove command.</value>
+		public RelayCommand Remove
+		{
+			get
+			{
+				if (this.remove is null)
+					this.remove = new RelayCommand(this.RemoveNumber, this.CanGetNumber);
+
+				return this.remove;
+			}
+			set
+			{
+				if (this.remove == value)
+					return;
+
+				this.remove = value;
+				this.RaisePropertyChanged(() => this.Remove);
+			}
+		}
+
+		/// <summary>
+		/// Removes the number from the collection.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <param name="obj">The object.</param>
+		private void RemoveNumber(object obj)
+		{
+			try
+			{
+				this.RestoreCurrentElement();
+				var removedItem = this.Collection.Remove();
+				this.Message = $"The item {removedItem} was removed.";
+			}
+			catch (InvalidOperationException ex)
+			{
+				this.Message = ex.Message;
+			}
+
+			if (this.Collection.Count() == 0)
+			{
+				this.Message += " The collection is empty.";
+				this.RestoreRemoveAndShowCommands();
+			}
+
+			this.RaisePropertyChanged(() => this.AllNumbers);
+		}
+
+		/// <summary>
+		/// Restores the current element.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private void RestoreCurrentElement()
+		{
+			if (this.CurrentElement.HasValue)
+				this.CurrentElement = null;
+		}
+
+		/// <summary>
+		/// Restores the remove and show commands.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private void RestoreRemoveAndShowCommands()
+		{
+			this.Show = null;
+			this.Remove = null;
+		}
+
+		/// <summary>
+		/// Gets or sets the selected type of the collection (array/linked list).
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <value>The type of the selected array linked list.</value>
+		public string SelectedArrayLinkedListType { get; set; }
```

Comment:
Please add default values for Types and Structures.
Also please change name to more informative to make possible to separate why we need two these types. Like BasicDataStructureTypes and DataStructureTypes.

Reply:
Names were changed to more understandable and according to the new enumeration.
But setting the default value to the appropriate collection types means that these types will use during configurating of dependencies. Therefore, user won't be able to change the main settings.

I'm about these dependencies
CollectionConfigCreator.InitializeCollection

Reply:
I meant that there are empty comboboxes on the UI and it is not good-looking.

Reply:
I suppose that empty combobox is not big deal. In that way user would be 100% percent sure that he chose the necessary collection type.

Reply:
The type None is default for the CustomCollectionType. The ui looks in this way now.
![image](https://user-images.githubusercontent.com/41588762/102223188-0b23d100-3eed-11eb-88cf-4b034f6660bc.png)